src/blocks/weather/met_no.rs

```diff
@@ -109,86 +111,93 @@ fn translate(legend: &LegendsStore, summary: &str, lang: &ApiLanguage) -> String
-        "lat" => lat,
-        "lon" => lon,
-        "altitude" => config.altitude.as_ref().unwrap(); if config.altitude.is_some()
-    };
-
-    let data: ForecastResponse = REQWEST_CLIENT
-        .get(FORECAST_URL)
-        .query(&querystr)
-        .header(reqwest::header::CONTENT_TYPE, "application/json")
-        .send()
-        .await
-        .error("Failed during request for current location")?
-        .json()
-        .await
-        .error("Failed while parsing location API result")?;
-
-    let first = &data.properties.timeseries.first().unwrap().data;
-
-    let instant = &first.instant.details;
-
-    let summary = first
-        .next_1_hours
-        .as_ref()
-        .unwrap()
-        .summary
-        .symbol_code
-        .split('_')
-        .next()
-        .unwrap();
-    let translated = translate(legend, summary, &config.lang);
-
-    Ok(WeatherResult {
-        location: "Unknown".to_string(),
-        temp: instant.air_temperature.unwrap_or_default(),
-        apparent: None,
-        humidity: instant.relative_humidity.unwrap_or_default(),
-        weather: translated.clone(),
-        weather_verbose: translated,
-        wind: instant.wind_speed.unwrap_or_default(),
-        wind_kmh: instant.wind_speed.unwrap_or_default() * 3.6,
-        wind_direction: convert_wind_direction(instant.wind_from_direction).into(),
-        icon: match summary {
-            "cloudy" | "partlycloudy" | "fair" | "fog" => WeatherIcon::Clouds,
-            "clearsky" => WeatherIcon::Sun,
-            "heavyrain" | "heavyrainshowers" | "lightrain" | "lightrainshowers" | "rain"
-            | "rainshowers" => WeatherIcon::Rain,
-            "rainandthunder"
-            | "heavyrainandthunder"
-            | "rainshowersandthunder"
-            | "sleetandthunder"
-            | "sleetshowersandthunder"
-            | "snowandthunder"
-            | "snowshowersandthunder"
-            | "heavyrainshowersandthunder"
-            | "heavysleetandthunder"
-            | "heavysleetshowersandthunder"
-            | "heavysnowandthunder"
-            | "heavysnowshowersandthunder"
-            | "lightsleetandthunder"
-            | "lightrainandthunder"
-            | "lightsnowandthunder"
-            | "lightssleetshowersandthunder"
-            | "lightssnowshowersandthunder"
-            | "lightrainshowersandthunder" => WeatherIcon::Thunder,
-            "heavysleet" | "heavysleetshowers" | "heavysnow" | "heavysnowshowers"
-            | "lightsleet" | "lightsleetshowers" | "lightsnow" | "lightsnowshowers" | "sleet"
-            | "sleetshowers" | "snow" | "snowshowers" => WeatherIcon::Snow,
-            _ => WeatherIcon::Default,
-        },
-    })
+#[async_trait]
+impl WeatherProvider for Config {
+    async fn get_weather(&mut self, location: &Option<LocationResponse>) -> Result<WeatherResult> {
+        if self.legend.is_none() {
+            self.legend = Some(get_legend().await?);
```

Comment:
This should be done in `new()` function. This way there will be no need for mutability & mutex.

Reply:
```rust
pub(super) struct Service {
    config: Config,
    legend: SomeType,
}

impl Service {
    pub(super) async fn new(config: Config) -> Result<Self> {
        todo!()
    }
}

#[async_trait]
impl WeatherProvider for Service {
    async fn get_weather(&self, location: &Option<LocationResponce>) -> Result<WeatherResult> {
        todo!()
    }
}
```

Reply:
Even better! Okay, implemented in afe4e23.